Add publish-new-version.sh for pushing versions of the crate

acmcarther opened the pull request:
Adds a script for publishing new versions of this crate to codify my shell history into a shell script usable by other folks.

Has some rudimentary checks for the existence of `cargo` and that the working directory is clean.

r: @UebelAndre as well

UebelAndre commented:
@acmcarther Are there also some examples of using this script? Whenever I add things like this to projects at work, I typically add a block of comments at the top of the file that show simple usage just to make it **crystal clear** as to what to do.
```bash
# Use the script like so
# Args: ...docs...
#
# ./my_script.sh arg1 arg2
```
```bash
# Use the script like so
# Environment: ...docs...
#
# ARG_VAR=1 ./my_other_script.sh
```
Things like that. 

While It's unlikely I'll be doing a deploy for this so If there's some internal Google doc that has this, that works as well. The thing I want to avoid is all maintainers who have familiarity with the project/script being unavailable and therefore no more releases happen because maybe some constraint changed or things are not as obvious to the available/new maintainers. 

Whether or not you decide to do that, I'm thankful for this PR! 😄

acmcarther commented:
Thanks, that's valuable feedback.

I've added some documentation about what the script does and how to use it to the top of the file. Incidentally, I've also updated the mechanism it uses to take in the new version from an environment variable to a regular argument.

UebelAndre commented on publish-new-version.sh:
```diff
@@ -0,0 +1,87 @@
+}
+
+cargo_build_crate() {
+  cargo build
```
Should this have `--release` with it?

acmcarther replied:
Done.

Added explanation for why we're doing this as well.

UebelAndre commented on publish-new-version.sh:
```diff
@@ -0,0 +1,87 @@
+PWD="$(pwd)"
+
+REPO_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$REPO_ROOT/impl"
```
How would you feel about `set -x` here? Might make logging of what the deploy process is doing nicer in the event of something going wrong.

acmcarther replied:
Now done!

UebelAndre commented:
@acmcarther When the version is updated should the [`Cargo.lock`](https://github.com/google/cargo-raze/blob/master/tools/Cargo.lock) file be updated too in the `/tools` directory?

acmcarther commented:
Re https://github.com/google/cargo-raze/pull/217#issuecomment-694394354
 
I'm not really sure. I don't consider the contents of `tools/` to be part of the package proper though, so i think it's OK to punt on this question for now.

That said, this should be a consideration in handling https://github.com/google/cargo-raze/issues/210 I think.

UebelAndre commented:
As a quick followup. Do you think you could delete some old branches? 😅

acmcarther commented:
Done!
